src/Api/Rest/Rest.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * @package     Redcore
+ * @subpackage  Api
+ *
+ * @copyright   Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Webservices\Api\Rest;
+
+use Joomla\Webservices\Api\ApiBase;
+use Joomla\Webservices\Resource\Item;
+use Joomla\Webservices\Resource\Resource;
+use Joomla\Webservices\Webservices\Webservice;
+use Joomla\Webservices\Webservices\Factory;
+
+use Joomla\DI\Container;
+use Joomla\Event\Event;
+use Joomla\Event\EventImmutable;
+use Joomla\Registry\Registry;
+
+/**
+ * Class to represent a REST interaction style.
+ *
+ * @since  1.2
+ */
+class Rest extends ApiBase
+{
+	/**
+	 * @var    string  Name of the Api
+	 * @since  1.2
+	 */
+	public $apiName = 'rest';
+
+	/**
+	 * Profile object.
+	 * 
+	 * @var  Profile
+	 */
+	private $profile = null;
+
+	/**
+	 * Main resource object
+	 * @var  Resource
+	 */
+	public $resource = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   Container  $container  The Dependency Injection Container object.
+	 * @param   Registry   $options    Optional custom options to load.
+	 *
+	 * @throws  \Exception
+	 * 
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(Container $container, Registry $options)
+	{
+		parent::__construct($container, $options);
+
+		// Set initial status code.
+		$this->setStatusCode(200);
+	}
+
+	/**
+	 * Execute the Api operation.
+	 *
+	 * @return  mixed  Webservice object with information on success, boolean false on failure.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  \Exception
+	 */
+	public function execute()
+	{
+		$options	= $this->getOptions();
+
+		$method		= $options->get('method', 'GET');
+		$task		= $options->get('task', '');
+		$format		= $options->get('format', '');
+		$clientName	= $options->get('webserviceClient');
+		$version	= $options->get('webserviceVersion');
+		$resourceName = $options->get('optionName');
+
+		// Map HTTP methods to service operations.
+		switch (strtolower($method))
+		{
+			case 'get':
+				$operation = !empty($task) ? 'task' : 'read';
+				break;
+
+			case 'put':
+				$operation = 'update';
+				break;
+
+			case 'post':
+				$operation = !empty($task) ? 'task' : 'create';
+				break;
+
+			case 'patch':
+				$operation = 'patch';
+				break;
+
+			case 'delete':
+				$operation = 'delete';
+				break;
+		}
+
+		$this->setOption('operation', $operation);
+
+		// If task is pointing to some other operation like apply, update or delete.
+//		if (!empty($task) && !empty($this->configuration->operations->task->{$task}['useOperation']))
```

Comment:
Ahh, I see what you're trying to do now.  My current thinking is that such a workaround should be restricted to just the POST method and it should be done through a new URL.  For example, if we're faced with /articles?long-string-of-search-criteria, instead of doing a POST to /articles?task=search with the criteria in the message body, we should instead do a POST to /articles/search or perhaps /articles/searchresults with the criteria in the message body.

Internally, we map the POST to "read" as you were doing, but if the situation makes it worthwhile, the POST could map to "create" (as it does by default) and result in new "searchcriteria" and "searchresults" resources being created.  The searchcriteria resource could support the full range of HTTP verbs.  The searchresults resource might only support GET.  This means that searches become cacheable and bookmarkable.  More thought needs to go into the search API and I have some ideas sketched out, but they need more work.

Apart from avoiding excessively long GET URIs, do you have any other use-cases where the "useOperation" field would be needed in REST interactions?